components/org.wso2.carbon.identity.authenticator.saml2.sso/src/main/java/org/wso2/carbon/identity/authenticator/saml2/sso/SAML2SSOAuthenticator.java

```diff
@@ -790,23 +790,30 @@ private String generatePassword(String username) {
                 for (Attribute attribute : attributesList) {
                     String attributeName = attribute.getName();
                     if (attributeName != null && roleClaim.equals(attributeName)) {
-                        // Assumes role claim appear only once
-                        Element value = attribute.getAttributeValues().get(0).getDOM();
-                        String attributeValue = value.getTextContent();
-
-                        if (log.isDebugEnabled()) {
-                            log.debug("AttributeName : " + attributeName + ", AttributeValue : " + attributeValue);
-                        }
-
-                        roles = attributeValue.split(getAttributeSeperator());
-                        if (log.isDebugEnabled()) {
-                            log.debug("Role list : " + Arrays.toString(roles));
+                        List<XMLObject> attributeValues = attribute.getAttributeValues();
+                        if (attributeValues != null && attributeValues.size() == 1) {
```

Comment:
The idea is role list can either come as a multiAttributeSeparator separated list in a single AttributeValue element or separately in multiple AttributeValue elements. So I don't think we need to consider having multiple roles in single AttributeValue if there are multiple AttributeValue elements